### PR TITLE
Add ability to manually adjust snap values in inspector

### PR DESCRIPTION
**Pull request by the-sink**
I thought it was fairly limiting to have hard-coded grid snap values, and wanted to implement a way to manually adjust them. I added a "gizmo snap values" section to the inspector at the bottom where you can adjust them. It saves the snap values to a file so they don't need to be manually re-entered after restarting the editor/reloading the plugin, and also means these values can be included in version control if a team so chooses.

image

Also didn't mess with it myself just yet but I was wondering if there was any objection to migrating the codebase over to the new Godot 4 syntax for signal connections, i.e.
```
plugin.selection_changed.connect(_on_selection_changed)
```

### CODE REVIEW

File: addons/ply/plugin.gd [hunk 1]
```diff
@@ -13,6 +13,8 @@ const Interop = preload("res://addons/ply/interop.gd")
 
 const PlyEditor = preload("res://addons/ply/nodes/ply.gd")
 
+const snap_data_path = "res://ply.dat"
```
Comment (jarneson) [inline addons/ply/plugin.gd:16]: we should use editor settings for this instead: https://docs.godotengine.org/en/latest/classes/class_editorsettings.html

**Review by jarneson — CHANGES_REQUESTED** [review]
amazing, one major change here to make it less noisy

File: addons/ply/plugin.gd (hunk 1 above)
Comment (the-sink) [reply addons/ply/plugin.gd]: changed in 8322262

not sure if I should remove the `snap_values` dict entirely or not - I wanted to stray away from doing `editor_settings.get_setting()` each time the Ctrl key was pressed, what do you think?

**Review by jarneson — APPROVED** [review]
nice

**Pull request merged**